src/site/apt/non-javac-compilers.apt.vm

```diff
@@ -35,29 +35,21 @@ Using Non-Javac Compilers
 
   The parameter <<<compilerId>>> determines which class will be used.
 
-  Currently, the Compiler Plugin is bundled with the <<<javac>>> compiler
-  with artifactId <<<plexus-compiler-javac>>> version <<${plexusCompilerVersion}>>,
-  which can be seen as one of the dependencies declared inside the Compiler Plugin's POM.
+  By default, the Compiler Plugin uses the <<<javac>>> compiler
+  bundled in the <<<javax.tools>>> package of the standard Java library.
+  But it is possible to use any other implementation, provided that the
+  implementation declares itself as a <<<javax.tools.JavaCompiler>>> service.
+  For using such implementation:
 
-  {{{https://codehaus-plexus.github.io/plexus-compiler/}Plexus Compiler}} component
-  has some other compiler Ids available under the groupId <<<org.codehaus.plexus>>>:
+  * Add the compiler implementation in the Maven compiler plugin dependencies.
 
-  * <<<aspectj>>> with artifactId <<<plexus-compiler-aspectj>>> - <<requires>>: <<<JDK 17+>>> and <<<Maven 3.9.6 +>>>
-
-  * <<<csharp>>> with artifactId <<<plexus-compiler-csharp>>> - <<requires>>: <<<JDK 8+>>>
-
-  * <<<eclipse>>> with artifactId <<<plexus-compiler-eclipse>>> - <<requires>>: <<<JDK 17+>>> and <<<Maven 3.9.6 +>>>
-
-  * <<<javac-with-errorprone>>>  with artifactId <<<plexus-compiler-javac-errorprone>>> - <<requires>>: <<<JDK 11+>>>
+  * Declare the compiler's identifier in the <<<compilerId>>> element of the Maven plugin configuration.
+    The identifier value shall be the value returned by the <<<JavaCompiler.name()>>> method of the implementation.
 
   []
 
-* Using The Other Plexus Compilers
-
-  To use any of the non-javac compilers, you need to make the artifact available
-  in the classpath when the Compiler Plugin runs. This is possible by adding the
-  dependency when declaring the plugin in your project's <<<pom.xml>>>. The example
-  below shows how to use the <<<csharp>>> compiler:
+  The example below shows how to use the Eclipse compiler <<<ecj>>>.
+  Note that at the contrary of Maven 3, there is no Plexus dependency to add.
```

Comment:
Why does Plexus pops up here?

Reply:
In Maven 3, it was necessary to add to dependency:

* The Eclipse compiler
* A Plexus extension which was a wrapper around the Eclipse compiler.

Actually, the Plexus extension had the Eclipse compiler as a transitive dependency, so it may look as a single dependency in the `pom.xml`.

In Maven 4, there is no longer a wrapper around the various compiler. The Eclipse compiler is used directly (through the standard `javax.tools.JavaCompiler` interface). I mentioned that in case some users were aware of the use of Plexus as an intermediate API, and were wondering whether Plexus was still needed. But I have no problem in omitting this details if too distracting.

Reply:
Removed the mention of Plexus.